org.lflang/src/org/lflang/generator/cpp/CppAssembleMethodGenerator.kt

```diff
@@ -123,14 +120,17 @@ class CppAssembleMethodGenerator(private val reactor: Reactor) {
     private fun setDeadline(reaction: Reaction): String =
         "${reaction.name}.set_deadline(${reaction.deadline.delay.toTime(true)}, [this]() { ${reaction.name}_deadline_handler(); });"
 
-    private fun assembleReaction(reaction: Reaction) = with(PrependOperator) {
-        """
+    private fun assembleReaction(reaction: Reaction): String {
+        val sources = reaction.sources.filter { it.variable is Port }
+        return with(PrependOperator) {
+            """
             |// ${reaction.name}
         ${" |"..reaction.triggers.joinToString(separator = "\n") { declareTrigger(reaction, it) }}
-        ${" |"..reaction.sources.joinToString(separator = "\n") { declareDependency(reaction, it) }}
+        ${" |"..sources.joinToString(separator = "\n") { declareDependency(reaction, it) }}
```

Comment:
Can one not use an action as a source? Or is it handled somewhere else?

Reply:
Yes, actions can be sources, but we don't need to tell this the runtime (the runtime only keeps track of the reaction that an action triggers). They are only brought into scope elsewhere.